### ISSUE TITLE
Allow specification of specific IP addresses when using proxy

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#1874]
After https://github.com/docker/docker/pull/19001, the incoming JSON on `create` can specify an IP address; we should accept this and pass it down in `WEAVE_CIDR`.

Related: #1734

**Comment by rade** [comment]
NB: the proxy probably needs to strip the option from the `create`, since otherwise Docker will either complain or try to do its own IP allocation.

**Comment by rade** [comment]
> the proxy probably needs to strip the option

That gives us no way to recover the aliases on restart. Unless we persist them ourselves.

**Comment by sheerun** [comment]
It seems to already work with docker plugin

**Comment by bboreham** [comment]
@sheerun yes; this was speculating about re-purposing the UI for our alternative option.
